**Bacharelado em Engenharia de Computação**

**Discente:** Paulo Guilherme Silva de Araújo - **Turma B**

Antes de começar a realizar as tarefas, será necessário importar o pandas.

In [1]:
import pandas as pd

**1° Tarefa**

Para carregar os DataFrames, pode-se usar o comando pd.read. O Pandas é capaz de ler diversas extensões de arquivos, como csv, json e html. A importação pode ser feita localmente, através dos arquivos do computador, ou através da internet. Para a realização das tarefas, irão ser importados os dados referentes aos anos de 2018, 2019 e 2020.

Para carregar os dados, nos argumentos do comando read.csv foram usados encoding='iso8859-1' e sep=';' para que o dataframe aparecesse de forma correta. Foi possível importar os dados diretamente do site, usando o argumento compression='zip'.

In [2]:
acidentes2020 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/jdDLrQIf33xXSCe/download', compression='zip',encoding='iso8859-1', sep=';')
acidentes2020

id data_inversa    dia_semana   horario  uf     br     km  \
0      260031.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1      260036.0   2020-01-01  quarta-feira  01:00:00  RJ  116.0    178   
2      260037.0   2020-01-01  quarta-feira  01:52:00  SC  101.0  206,9   
3      260038.0   2020-01-01  quarta-feira  01:15:00  RO  364.0    236   
4      260039.0   2020-01-01  quarta-feira  01:30:00  PR  153.0    360   
...         ...          ...           ...       ...  ..    ...    ...   
63543  364615.0   2020-03-20   sexta-feira  09:00:00  RN  304.0  171,8   
63544  364810.0   2020-09-12        sábado  23:50:00  BA  116.0   11,5   
63545  364817.0   2020-10-30   sexta-feira  10:30:00  BA  116.0   14,3   
63546  364835.0   2020-09-20       domingo  23:00:00  BA  116.0      8   
63547  364903.0   2020-10-31        sábado  02:10:00  BA  116.0      3   

                  municipio                causa_acidente  \
0                    GURUPI              Animais na Pista   
1               NOVA IGUACU  Falta de Atenção do Pedestre   
2                  SAO JOSE            Ingestão de Álcool   
3                    CACOAL       Velocidade Incompatível   
4                  REBOUCAS            Ingestão de Álcool   
...                     ...                           ...   
63543               ANGICOS   Falta de Atenção à Condução   
63544  VITORIA DA CONQUISTA            Ingestão de Álcool   
63545  VITORIA DA CONQUISTA   Falta de Atenção à Condução   
63546  VITORIA DA CONQUISTA       Velocidade Incompatível   
63547  VITORIA DA CONQUISTA              Animais na Pista   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0           Atropelamento de Animal  ...              5      0         0   
1         Atropelamento de Pedestre  ...              0      1         0   
2                   Colisão lateral  ...              0      1         0   
3         Saída de leito carroçável  ...              1      0         0   
4                       Capotamento  ...              4      0         0   
...                             ...  ...            ...    ...       ...   
63543              Colisão traseira  ...              1      2         0   
63544               Colisão lateral  ...              0      2         0   
63545               Colisão lateral  ...              0      2         0   
63546  Queda de ocupante de veículo  ...              0      0         0   
63547       Atropelamento de Animal  ...              1      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           5        1  -11,77460203  -49,10744996     SR-TO   UOP01/TO   
1           0        1  -22,75223028   -43,4379103     SR-RJ     DEL5/1   
2           1        2  -27,59193546  -48,61824557     SR-SC     DEL8/1   
3           1        1  -11,44624577  -61,43761218     SR-RO    DEL21/2   
4           6        1  -25,67503796  -50,75089805     SR-PR     DEL7/3   
...       ...      ...           ...           ...       ...        ...   
63543       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
63544       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
63545       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
63546       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
63547       1        1    -14,824848    -40,861311     SR-BA      UOP01   

            uop  
0           NaN  
1      UOP02/RJ  
2      UOP01/SC  
3      UOP02/RO  
4      UOP04/PR  
...         ...  
63543     DEL01  
63544     DEL08  
63545     DEL08  
63546     DEL08  
63547     DEL08  

[63548 rows x 30 columns]

In [4]:
acidentes2019 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/kRBUylqz6DyQznN/download', compression='zip',encoding='iso8859-1', sep=';')
acidentes2019

id data_inversa     dia_semana   horario  uf     br     km  \
0      182210.0   2019-01-01    terça-feira  01:30:00  SP  116.0    218   
1      182211.0   2019-01-01    terça-feira  01:30:00  PR  373.0  177,3   
2      182212.0   2019-01-01    terça-feira  00:45:00  SC  101.0     16   
3      182214.0   2019-01-01    terça-feira  00:00:00  CE   20.0  429,5   
4      182215.0   2019-01-01    terça-feira  01:00:00  MG  381.0    327   
...         ...          ...            ...       ...  ..    ...    ...   
67441  266255.0   2019-10-07  segunda-feira  09:10:00  ES  101.0  270,5   
67442  266406.0   2019-11-25  segunda-feira  07:20:00  PR  116.0    125   
67443  266434.0   2019-06-03  segunda-feira  19:00:00  CE  222.0    1,4   
67444  266573.0   2019-07-13         sábado  19:35:00  PR  373.0    425   
67445  266627.0   2019-10-22    terça-feira  20:20:00  BA  324.0    608   

                municipio                       causa_acidente  \
0               GUARULHOS          Falta de Atenção à Condução   
1            PONTA GROSSA          Falta de Atenção à Condução   
2                  GARUVA                     Animais na Pista   
3               FORTALEZA  Ingestão de Substâncias Psicoativas   
4                NOVA ERA          Falta de Atenção à Condução   
...                   ...                                  ...   
67441               SERRA          Falta de Atenção à Condução   
67442  FAZENDA RIO GRANDE          Falta de Atenção à Condução   
67443             CAUCAIA   Não guardar distância de segurança   
67444              CANDOI          Defeito Mecânico no Veículo   
67445        SIMOES FILHO                       Defeito na Via   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0       Colisão com objeto estático  ...              0      0         0   
1                  Colisão traseira  ...              0      0         0   
2       Colisão com objeto estático  ...              0      0         0   
3       Colisão com objeto estático  ...              1      0         0   
4               Colisão transversal  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
67441                    Tombamento  ...              1      1         0   
67442               Colisão lateral  ...              0      1         1   
67443              Colisão traseira  ...              1      1         0   
67444     Saída de leito carroçável  ...              1      0         0   
67445  Queda de ocupante de veículo  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1  -23,46052014  -46,48772478     SR-SP     DEL6/1   
1           1        1  -25,05533957  -50,22776753     SR-PR     DEL7/3   
2           1        1      -26,1216      -48,8826     SR-SC     DEL8/3   
3           1        1      -3,76999    -38,670063     SR-CE    DEL16/1   
4           1        2      -19,7609      -43,0306     SR-MG     DEL4/3   
...       ...      ...           ...           ...       ...        ...   
67441       1        2     -20,23107     -40,27387     SR-ES    DEL12/2   
67442       1        2     -25,59505     -49,31631     SR-PR     DEL7/1   
67443       1        2     -3,736507     -38,65337     SR-CE    DEL16/1   
67444       1        1      -25,7118     -52,18374     SR-PR     DEL7/3   
67445       1        1     -12,78156     -38,41656     SR-BA    DEL10/1   

            uop  
0      UOP01/SP  
1      UOP01/PR  
2      UOP01/SC  
3      UOP02/CE  
4      UOP01/MG  
...         ...  
67441  UOP01/ES  
67442  UOP03/PR  
67443  UOP01/CE  
67444  UOP02/PR  
67445  UOP01/BA  

[67446 rows x 30 columns]

In [3]:
acidentes2018 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/MaC6cieXSFACNWT/download', compression='zip', encoding='iso8859-1', sep=';')
acidentes2018

id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
69290  252256.0   2018-12-24  segunda-feira  09:15:00  PI  343.0  263,1   
69291  252652.0   2018-12-12   quarta-feira  07:00:00  BA  101.0    321   
69292  252801.0   2018-11-15   quinta-feira  10:48:00  SP  116.0    529   
69293  253040.0   2018-11-23    sexta-feira  19:30:00  RS  472.0  573,5   
69294  253078.0   2018-10-01  segunda-feira  04:31:00  GO   40.0      9   

                       municipio  \
0                        RESENDE   
1                  FLORIANOPOLIS   
2                       ITABORAI   
3                        SARANDI   
4                       CANDIOTA   
...                          ...   
69290                CAMPO MAIOR   
69291  PRESIDENTE TANCREDO NEVES   
69292             BARRA DO TURVO   
69293                 URUGUAIANA   
69294                   LUZIANIA   

                                          causa_acidente  \
0                                      Condutor Dormindo   
1                     Não guardar distância de segurança   
2                                 Ultrapassagem Indevida   
3                                     Ingestão de Álcool   
4                            Falta de Atenção à Condução   
...                                                  ...   
69290                                   Agressão Externa   
69291                        Falta de Atenção à Condução   
69292                        Falta de Atenção à Condução   
69293  Desobediência às normas de trânsito pelo condutor   
69294                 Não guardar distância de segurança   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
69290   Colisão com objeto estático  ...              0      1         0   
69291     Saída de leito carroçável  ...              0      0         0   
69292              Colisão traseira  ...              0      0         1   
69293  Queda de ocupante de veículo  ...              1      1         0   
69294     Saída de leito carroçável  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
69290       0        1     -4,819326    -42,158821     SR-PI    DEL17/1   
69291       1        1  -13,44768518  -39,42066193     SR-BA    DEL10/5   
69292       1        2    -24,939211    -48,301248     SR-SP     DEL6/5   
69293       1        2  -29,76355779  -57,05200195     SR-RS    DEL9/13   
69294       1        1    -16,131158    -47,957739     SR-DF   UOP01/DF   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
69290  UOP03/PI  


**2° Tarefa**

**a)** Para saber a quantidade de acidentes, basta verificar a quantidade de linhas excluindo dados faltosos. Para isso, pode-se usar o comando value_counts() e sum()

In [6]:
acidentes2020['id'].value_counts().sum()

63548

In [7]:
acidentes2019['id'].value_counts().sum()

67446

In [8]:
acidentes2018['id'].value_counts().sum()

69295

Portanto, pode-se perceber que:

Em 2020 houveram 63548 acidentes 

Em 2019 houveram 67446 acidentes 

Em 2018 houveram 69295 acidentes

**b)** Através do comando .info(), é possível identificar que cada conjunto de dados possui 30 variáveis

In [ ]:
acidentes2020.info()

In [ ]:
acidentes2019.info()

In [ ]:
acidentes2018.info()

**c)** Para resolver essa questão, vou filtrar as colunas colunas "municipio" e "br", juntar todos os 3 anos para fazer a contagem e organizar em ordem decrescente com o comando value_counts().

In [9]:
acidentesbr2020 = acidentes2020[['municipio','br']]
acidentesbr2019 = acidentes2019[['municipio','br']]
acidentesbr2018 = acidentes2018[['municipio','br']]

In [10]:
acidentesbr = pd.concat([acidentesbr2020, acidentesbr2019, acidentesbr2018], axis=0)
acidentesbr['municipio'].value_counts()

CURITIBA               3063
BRASILIA               2997
SAO JOSE               2322
GUARULHOS              2111
PALHOCA                1887
                       ... 
IPANGUACU                 1
TIROS                     1
SANTA CRUZ CABRALIA       1
CRUZILIA                  1
NATIVIDADE                1
Name: municipio, Length: 1926, dtype: int64

A partir desse comando, foi possível ver que os 5 municípios com mais acidentes em rodovias federais são:

Curitiba, Brasília, São José, Guarulhos e Palhoça

**d)** Para saber quantos feridos graves na Paraíba, podemos realizar uma filtragem na coluna "feridos_graves" e na coluna "uf". Depois, pode-se chamar as funções value_counts() para contar os valores e sum() para somar todos.

In [17]:
feridosPB = acidentes2019[(acidentes2019['uf'] == 'PB')&(acidentes2019['feridos_graves'] >= 1)]
feridosPB['feridos_graves'].value_counts().sum()

414

Chegando assim ao resultado de 414 feridos graves na Paraíba em 2019

**3° Tarefa**

**a)** Para fazer um ranking com os acidentes por estado, pode-se usar o comando value_counts(). Antes, irei filtrar as informações que eu quero e juntar os conjuntos de dados de todos os anos. Além do value_counts, usei o sort_values para organizar os dados.

Filtrei as informações de uf e tipo_acidente do dataframe original para criar o ranking.

In [23]:
ranking2020 = acidentes2020[['uf', 'tipo_acidente']]
ranking2019 = acidentes2019[['uf', 'tipo_acidente']]
ranking2018 = acidentes2018[['uf', 'tipo_acidente']]

Depois, com o value_counts, e sort_values, organizei os dados do maior para o menor, mostrando os estados com mais acidentes e menos acidentes de cada ano.

In [24]:
data20=ranking2020.value_counts().sort_values(ascending=False)
data19=ranking2019.value_counts().sort_values(ascending=False)
data18=ranking2018.value_counts().sort_values(ascending=False)

Depois, joguei os dados em outro dataframe para fazer a concatenação. Além disso, resetei os indices com reset_index(), renomeei a coluna das quantidades de acidentes e criei uma nova coluna com os respectivos anos. Isso é importante para fazer a apresentação dos dados de forma correta e também realizar o multindex.

In [66]:
df20=pd.DataFrame(data=data20).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2020)
df19=pd.DataFrame(data=data19).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2019)
df18=pd.DataFrame(data=data18).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2018)

Por fim, setei os indices para fazer o multindex e criei um dataframe final concatenando os dataframes de 2020, 2019 e 2018.

In [67]:
df20.set_index(['ano', 'uf'], inplace=True)
df19.set_index(['ano', 'uf'], inplace=True)
df18.set_index(['ano', 'uf'], inplace=True)

In [68]:
ranking = pd.concat([df20, df19, df18], axis=0)
ranking

tipo_acidente  quantidade
ano  uf                                          
2020 MG     Saída de leito carroçável        1839
     PR              Colisão traseira        1414
     PR     Saída de leito carroçável        1399
     SC              Colisão traseira        1368
     MG              Colisão traseira        1279
...                               ...         ...
2018 AC               Danos eventuais           1
     AP  Queda de ocupante de veículo           1
     SE         Derramamento de carga           1
     AM               Danos eventuais           1
     RN         Derramamento de carga           1

[1295 rows x 2 columns]

In [69]:
ranking.to_csv('ranking3a.csv')

**b)** Para o DataFrame dos dias da semana, pode-se seguir a mesma lógica da questão anterior. A diferença é que, nesse caso, não é necessário usar o sort_values() para fazer a ordenação. Além disso, na função value_counts, usei o argumento sort=False pois não havia a necessidade de organizar os itens em ranking. Exceto isso, a ordem de comandos usados manteve-se

In [28]:
semana2020 = acidentes2020[['dia_semana', 'tipo_acidente']]
semana2019 = acidentes2020[['dia_semana', 'tipo_acidente']]
semana2018 = acidentes2020[['dia_semana', 'tipo_acidente']]

In [29]:
sem20=semana2020.value_counts(sort=False)
sem19=semana2019.value_counts(sort=False)
sem18=semana2018.value_counts(sort=False)

In [30]:
dfs20=pd.DataFrame(data=sem20).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2020)
dfs19=pd.DataFrame(data=sem19).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2019)
dfs18=pd.DataFrame(data=sem18).reset_index().rename(columns={0: 'quantidade'}).assign(ano=2018)

In [31]:
dfs20.set_index(['ano', 'dia_semana'], inplace=True)
dfs19.set_index(['ano', 'dia_semana'], inplace=True)
dfs18.set_index(['ano', 'dia_semana'], inplace=True)

In [32]:
dfsemana = pd.concat([dfs20, dfs19, dfs18], axis=0)

In [33]:
dfsemana

tipo_acidente  quantidade
ano  dia_semana                                              
2020 domingo              Atropelamento de Animal         169
     domingo            Atropelamento de Pedestre         428
     domingo                          Capotamento         415
     domingo                   Colisão com objeto           1
     domingo      Colisão com objeto em movimento          26
...                                           ...         ...
2018 terça-feira                    Engavetamento         137
     terça-feira                         Incêndio         159
     terça-feira     Queda de ocupante de veículo         316
     terça-feira        Saída de leito carroçável        1134
     terça-feira                       Tombamento         750

[390 rows x 2 columns]

In [70]:
dfsemana.to_csv('dfsemana3b.csv')

**4° Tarefa**

**a)** Para sabermos a causa mais e menos frequente de acidente, podemos juntar as causas de todos os anos e usar o comando value_counts(), pois ele retorna todos esses valores. Após a contagem dos anos separadamente, somei usando o comando sum().

In [37]:
c1 = acidentes2020['causa_acidente'].value_counts()
c2 = acidentes2019['causa_acidente'].value_counts()
c3 = acidentes2018['causa_acidente'].value_counts()

In [38]:
causas = pd.DataFrame(data = [c1, c2, c3])

In [39]:
causas.sum(axis=0)

Falta de Atenção à Condução                                                       73231.0
Desobediência às normas de trânsito pelo condutor                                 24089.0
Velocidade Incompatível                                                           18763.0
Ingestão de Álcool                                                                15689.0
Defeito Mecânico no Veículo                                                       11190.0
Não guardar distância de segurança                                                12082.0
Condutor Dormindo                                                                  7225.0
Pista Escorregadia                                                                 6752.0
Falta de Atenção do Pedestre                                                       5685.0
Animais na Pista                                                                   4551.0
Ultrapassagem Indevida                                                             3590.0
Defeito na

Sendo assim, podemos ver que a causa mais frequente de acidentes é **Falta de Atenção à Condução** e a menos frequente é **Ausência de sinalização**

**b)** Para a realização dessa tarefa, usarei o DatetimeIndex para criar uma coluna com os meses. Isso é necessário para separar as datas dos DataFrames em meses.

In [40]:
acidentes2020['mes'] = pd.DatetimeIndex(acidentes2020['data_inversa']).month
acidentes2019['mes'] = pd.DatetimeIndex(acidentes2019['data_inversa']).month
acidentes2018['mes'] = pd.DatetimeIndex(acidentes2018['data_inversa']).month

Após isso, irei separar as colunas de meses, ilesos, feridos_graves e pessoas em outro DF e usar o sort_values para ordená-los, para que, logo após, seja feita o cálculo de proporção usando o sum() e depois dividindo, usando a seguinte equação: **numero de ilesos ou feridos graves/ número total de pessoas envolvidas no acidente** Isso será feito em cada mês do ano, para os 3 anos.

In [41]:
dfprop20 = acidentes2020[['mes', 'ilesos', 'feridos_graves', 'pessoas']].sort_values(by='mes')
dfprop19 = acidentes2019[['mes', 'ilesos', 'feridos_graves', 'pessoas']].sort_values(by='mes')
dfprop18 = acidentes2018[['mes', 'ilesos', 'feridos_graves', 'pessoas']].sort_values(by='mes')

In [42]:
jan20 = dfprop20[dfprop20['mes']==1]
print('Jan/20') 
print('ilesos/pessoas:', jan20['ilesos'].sum()/jan20['pessoas'].sum())
print('feridos graves/pessoas: ', jan20['feridos_graves'].sum()/jan20['pessoas'].sum())
fev20 = dfprop20[dfprop20['mes']==2]
print('Fev/20') 
print('ilesos/pessoas:', fev20['ilesos'].sum()/fev20['pessoas'].sum())
print('feridos graves/pessoas: ', fev20['feridos_graves'].sum()/fev20['pessoas'].sum())
mar20 = dfprop20[dfprop20['mes']==3]
print('Mar/20') 
print('ilesos/pessoas:', mar20['ilesos'].sum()/mar20['pessoas'].sum())
print('feridos graves/pessoas: ', mar20['feridos_graves'].sum()/mar20['pessoas'].sum())
abr20 = dfprop20[dfprop20['mes']==4]
print('Abr/20') 
print('ilesos/pessoas:', abr20['ilesos'].sum()/abr20['pessoas'].sum())
print('feridos graves/pessoas: ', abr20['feridos_graves'].sum()/abr20['pessoas'].sum())
mai20 = dfprop20[dfprop20['mes']==5]
print('Mai/20') 
print('ilesos/pessoas:', mai20['ilesos'].sum()/mai20['pessoas'].sum())
print('feridos graves/pessoas: ', mai20['feridos_graves'].sum()/mai20['pessoas'].sum())
jun20 = dfprop20[dfprop20['mes']==6]
print('Jun/20') 
print('ilesos/pessoas:', jun20['ilesos'].sum()/jun20['pessoas'].sum())
print('feridos graves/pessoas: ', jun20['feridos_graves'].sum()/jun20['pessoas'].sum())
jul20 = dfprop20[dfprop20['mes']==7]
print('Jul/20') 
print('ilesos/pessoas:', jul20['ilesos'].sum()/jul20['pessoas'].sum())
print('feridos graves/pessoas: ', jul20['feridos_graves'].sum()/jul20['pessoas'].sum())
ago20 = dfprop20[dfprop20['mes']==8]
print('Ago/20') 
print('ilesos/pessoas:', ago20['ilesos'].sum()/ago20['pessoas'].sum())
print('feridos graves/pessoas: ', ago20['feridos_graves'].sum()/ago20['pessoas'].sum())
set20 = dfprop20[dfprop20['mes']==9]
print('Set/20') 
print('ilesos/pessoas:', set20['ilesos'].sum()/set20['pessoas'].sum())
print('feridos graves/pessoas: ', set20['feridos_graves'].sum()/set20['pessoas'].sum())
out20 = dfprop20[dfprop20['mes']==10]
print('Out/20') 
print('ilesos/pessoas:', out20['ilesos'].sum()/out20['pessoas'].sum())
print('feridos graves/pessoas: ', out20['feridos_graves'].sum()/out20['pessoas'].sum())
nov20 = dfprop20[dfprop20['mes']==11]
print('Nov/20') 
print('ilesos/pessoas:', nov20['ilesos'].sum()/nov20['pessoas'].sum())
print('feridos graves/pessoas: ', nov20['feridos_graves'].sum()/nov20['pessoas'].sum())
dez20 = dfprop20[dfprop20['mes']==12]
print('Dez/20') 
print('ilesos/pessoas:', dez20['ilesos'].sum()/dez20['pessoas'].sum())
print('feridos graves/pessoas: ', dez20['feridos_graves'].sum()/dez20['pessoas'].sum())

Jan/20
ilesos/pessoas: 0.4155650319829424
feridos graves/pessoas:  0.10746268656716418
Fev/20
ilesos/pessoas: 0.4200106197375408
feridos graves/pessoas:  0.11135553364181143
Mar/20
ilesos/pessoas: 0.4010208816705336
feridos graves/pessoas:  0.1125754060324826
Abr/20
ilesos/pessoas: 0.37626714370900416
feridos graves/pessoas:  0.11794871794871795
Mai/20
ilesos/pessoas: 0.40133015683938855
feridos graves/pessoas:  0.11594202898550725
Jun/20
ilesos/pessoas: 0.4232829538932466
feridos graves/pessoas:  0.11593648364262484
Jul/20
ilesos/pessoas: 0.42432060543515654
feridos graves/pessoas:  0.1086171310629515
Ago/20
ilesos/pessoas: 0.42341433118546695
feridos graves/pessoas:  0.11916776647775794
Set/20
ilesos/pessoas: 0.41601577310988097
feridos graves/pessoas:  0.11738833699855919
Out/20
ilesos/pessoas: 0.4110061375077581
feridos graves/pessoas:  0.11971588166333356
Nov/20
ilesos/pessoas: 0.4070422535211268
feridos graves/pessoas:  0.12038547071905115
Dez/20
ilesos/pessoas: 0.412292358803986

In [43]:
jan19 = dfprop19[dfprop19['mes']==1]
print('Jan/19') 
print('ilesos/pessoas:', jan19['ilesos'].sum()/jan19['pessoas'].sum())
print('feridos graves/pessoas: ', jan19['feridos_graves'].sum()/jan19['pessoas'].sum())
fev19 = dfprop19[dfprop19['mes']==2]
print('Fev/19') 
print('ilesos/pessoas:', fev19['ilesos'].sum()/fev19['pessoas'].sum())
print('feridos graves/pessoas: ', fev19['feridos_graves'].sum()/fev19['pessoas'].sum())
mar19 = dfprop19[dfprop19['mes']==3]
print('Mar/19') 
print('ilesos/pessoas:', mar19['ilesos'].sum()/mar19['pessoas'].sum())
print('feridos graves/pessoas: ', mar19['feridos_graves'].sum()/mar19['pessoas'].sum())
abr19 = dfprop19[dfprop19['mes']==4]
print('Abr/19') 
print('ilesos/pessoas:', abr19['ilesos'].sum()/abr19['pessoas'].sum())
print('feridos graves/pessoas: ', abr19['feridos_graves'].sum()/abr19['pessoas'].sum())
mai19 = dfprop19[dfprop19['mes']==5]
print('Mai/19') 
print('ilesos/pessoas:', mai19['ilesos'].sum()/mai19['pessoas'].sum())
print('feridos graves/pessoas: ', mai19['feridos_graves'].sum()/mai19['pessoas'].sum())
jun19 = dfprop19[dfprop19['mes']==6]
print('Jun/19') 
print('ilesos/pessoas:', jun19['ilesos'].sum()/jun19['pessoas'].sum())
print('feridos graves/pessoas: ', jun19['feridos_graves'].sum()/jun19['pessoas'].sum())
jul19 = dfprop19[dfprop19['mes']==7]
print('Jul/19') 
print('ilesos/pessoas:', jul19['ilesos'].sum()/jul19['pessoas'].sum())
print('feridos graves/pessoas: ', jul19['feridos_graves'].sum()/jul19['pessoas'].sum())
ago19 = dfprop19[dfprop19['mes']==8]
print('Ago/19') 
print('ilesos/pessoas:', ago19['ilesos'].sum()/ago19['pessoas'].sum())
print('feridos graves/pessoas: ', ago19['feridos_graves'].sum()/ago19['pessoas'].sum())
set19 = dfprop19[dfprop19['mes']==9]
print('Set/19') 
print('ilesos/pessoas:', set19['ilesos'].sum()/set19['pessoas'].sum())
print('feridos graves/pessoas: ', set19['feridos_graves'].sum()/set19['pessoas'].sum())
out19 = dfprop19[dfprop19['mes']==10]
print('Out/19') 
print('ilesos/pessoas:', out19['ilesos'].sum()/out19['pessoas'].sum())
print('feridos graves/pessoas: ', out19['feridos_graves'].sum()/out19['pessoas'].sum())
nov19 = dfprop19[dfprop19['mes']==11]
print('Nov/19') 
print('ilesos/pessoas:', nov19['ilesos'].sum()/nov19['pessoas'].sum())
print('feridos graves/pessoas: ', nov19['feridos_graves'].sum()/nov19['pessoas'].sum())
dez19 = dfprop19[dfprop19['mes']==12]
print('Dez/19') 
print('ilesos/pessoas:', dez19['ilesos'].sum()/dez19['pessoas'].sum())
print('feridos graves/pessoas: ', dez19['feridos_graves'].sum()/dez19['pessoas'].sum())

Jan/19
ilesos/pessoas: 0.42422393410113646
feridos graves/pessoas:  0.11326367172603158
Fev/19
ilesos/pessoas: 0.4241037376048818
feridos graves/pessoas:  0.10678871090770405
Mar/19
ilesos/pessoas: 0.4331933082762178
feridos graves/pessoas:  0.11459945463925124
Abr/19
ilesos/pessoas: 0.4156847444401092
feridos graves/pessoas:  0.10596956691377292
Mai/19
ilesos/pessoas: 0.4151621727237202
feridos graves/pessoas:  0.12075029308323564
Jun/19
ilesos/pessoas: 0.4237566177018865
feridos graves/pessoas:  0.12161658340168519
Jul/19
ilesos/pessoas: 0.43681529551322307
feridos graves/pessoas:  0.11667015560672668
Ago/19
ilesos/pessoas: 0.413083972878325
feridos graves/pessoas:  0.11668281052082558
Set/19
ilesos/pessoas: 0.4210641719046923
feridos graves/pessoas:  0.11065633679286654
Out/19
ilesos/pessoas: 0.41892182901442826
feridos graves/pessoas:  0.11980475199196038
Nov/19
ilesos/pessoas: 0.425812052376474
feridos graves/pessoas:  0.11104680604789119
Dez/19
ilesos/pessoas: 0.4222649983958935


In [44]:
jan18 = dfprop18[dfprop18['mes']==1]
print('Jan/18') 
print('ilesos/pessoas:', jan18['ilesos'].sum()/jan18['pessoas'].sum())
print('feridos graves/pessoas: ', jan18['feridos_graves'].sum()/jan18['pessoas'].sum())
fev18 = dfprop18[dfprop18['mes']==2]
print('Fev/18') 
print('ilesos/pessoas:', fev18['ilesos'].sum()/fev18['pessoas'].sum())
print('feridos graves/pessoas: ', fev18['feridos_graves'].sum()/fev18['pessoas'].sum())
mar18 = dfprop18[dfprop18['mes']==3]
print('Mar/18') 
print('ilesos/pessoas:', mar18['ilesos'].sum()/mar18['pessoas'].sum())
print('feridos graves/pessoas: ', mar18['feridos_graves'].sum()/mar18['pessoas'].sum())
abr18 = dfprop18[dfprop18['mes']==4]
print('Abr/18') 
print('ilesos/pessoas:', abr18['ilesos'].sum()/abr18['pessoas'].sum())
print('feridos graves/pessoas: ', abr18['feridos_graves'].sum()/abr18['pessoas'].sum())
mai18 = dfprop18[dfprop18['mes']==5]
print('Mai/18') 
print('ilesos/pessoas:', mai18['ilesos'].sum()/mai18['pessoas'].sum())
print('feridos graves/pessoas: ', mai18['feridos_graves'].sum()/mai18['pessoas'].sum())
jun18 = dfprop18[dfprop18['mes']==6]
print('Jun/18') 
print('ilesos/pessoas:', jun18['ilesos'].sum()/jun18['pessoas'].sum())
print('feridos graves/pessoas: ', jun18['feridos_graves'].sum()/jun18['pessoas'].sum())
jul18 = dfprop18[dfprop18['mes']==7]
print('Jul/18') 
print('ilesos/pessoas:', jul18['ilesos'].sum()/jul18['pessoas'].sum())
print('feridos graves/pessoas: ', jul18['feridos_graves'].sum()/jul18['pessoas'].sum())
ago18 = dfprop18[dfprop18['mes']==8]
print('Ago/18') 
print('ilesos/pessoas:', ago18['ilesos'].sum()/ago18['pessoas'].sum())
print('feridos graves/pessoas: ', ago18['feridos_graves'].sum()/ago18['pessoas'].sum())
set18 = dfprop18[dfprop18['mes']==9]
print('Set/18') 
print('ilesos/pessoas:', set18['ilesos'].sum()/set18['pessoas'].sum())
print('feridos graves/pessoas: ', set18['feridos_graves'].sum()/set18['pessoas'].sum())
out18 = dfprop18[dfprop18['mes']==10]
print('Out/18') 
print('ilesos/pessoas:', out18['ilesos'].sum()/out18['pessoas'].sum())
print('feridos graves/pessoas: ', out18['feridos_graves'].sum()/out18['pessoas'].sum())
nov18 = dfprop18[dfprop18['mes']==11]
print('Nov/18') 
print('ilesos/pessoas:', nov18['ilesos'].sum()/nov18['pessoas'].sum())
print('feridos graves/pessoas: ', nov18['feridos_graves'].sum()/nov18['pessoas'].sum())
dez18 = dfprop18[dfprop18['mes']==12]
print('Dez/18') 
print('ilesos/pessoas:', dez18['ilesos'].sum()/dez18['pessoas'].sum())
print('feridos graves/pessoas: ', dez18['feridos_graves'].sum()/dez18['pessoas'].sum())

Jan/18
ilesos/pessoas: 0.5132913291329133
feridos graves/pessoas:  0.09060906090609061
Fev/18
ilesos/pessoas: 0.49825585534277783
feridos graves/pessoas:  0.09219050331031536
Mar/18
ilesos/pessoas: 0.4815354070152217
feridos graves/pessoas:  0.10185307743216412
Abr/18
ilesos/pessoas: 0.4301587301587302
feridos graves/pessoas:  0.1127739984882842
Mai/18
ilesos/pessoas: 0.42222995909130473
feridos graves/pessoas:  0.11445730698929411
Jun/18
ilesos/pessoas: 0.4250971250971251
feridos graves/pessoas:  0.1163947163947164
Jul/18
ilesos/pessoas: 0.4375091790277574
feridos graves/pessoas:  0.11029519753267734
Ago/18
ilesos/pessoas: 0.42705072010018785
feridos graves/pessoas:  0.1063713212273012
Set/18
ilesos/pessoas: 0.4206647731634298
feridos graves/pessoas:  0.11545252168572964
Out/18
ilesos/pessoas: 0.43017843289371605
feridos graves/pessoas:  0.1126454615981381
Nov/18
ilesos/pessoas: 0.43335606577252406
feridos graves/pessoas:  0.11540501629158142
Dez/18
ilesos/pessoas: 0.43291569602719887

**c)** Para sabermos se a pandemia afetou os acidentes em rodovias federais, podemos comparar a quantidade de acidentes, no período de março a dezembro, de 2020 com o mesmo período nos anos anteriores. Para isso, realizarei uma filtragem nos dataframes e somarei os acidentes do período para compará-los.

Acidentes de Março a Dezembro de 2018:

In [45]:
(acidentes2018[acidentes2018['mes']>=3].value_counts('tipo_acidente').sum())

56163

Acidentes de Março a Dezembro de 2019:

In [46]:
(acidentes2019[acidentes2019['mes']>=3].value_counts('tipo_acidente').sum())

57363

Acidentes de Março a Dezembro de 2020:

In [47]:
(acidentes2020[acidentes2020['mes']>=3].value_counts('tipo_acidente').sum())

52584

Sendo assim, é possível ver que houve uma diminuição significativa de acidentes no período, comparada com a mesma quantidade de acidentes dos anos anteriores.